docs/standard/serialization/system-text-json-how-to.md

```diff
@@ -0,0 +1,768 @@
+
+|Property |Value  |Notes  |
+|---------|---------|---------|
+| Date    | 8/1/2019 12:00:00 AM -07:00||
+| TemperatureC| 0 | Case-sensitive mismatch (`temperatureC` in the JSON), so the property isn't set. |
+| Summary | Hot ||
+| ExtensionData | temperatureC: 25 |Since the case didn't match, this JSON property is an extra and becomes a key-value pair in the dictionary.|
+|| DatesAvailable:<br>  8/1/2019 12:00:00 AM -07:00<br>8/2/2019 12:00:00 AM -07:00 |Extra property from the JSON becomes a key-value pair, with an array as the value object.|
+| |SummaryWords:<br>Cool<br>Windy<br>Humid |Extra property from the JSON becomes a key-value pair, with an array as the value object.|
+
+When the target object is serialized, the extension data key value pairs become JSON properties just as they were in the incoming JSON:
+
+```json
+{
+  "Date": "2019-08-01T00:00:00-07:00",
+  "TemperatureC": 0,
+  "Summary": "Hot",
+  "temperatureC": 25,
+  "DatesAvailable": [
+    "2019-08-01T00:00:00-07:00",
+    "2019-08-02T00:00:00-07:00"
+  ],
+  "SummaryWords": [
+    "Cool",
+    "Windy",
+    "Humid"
+  ]
+}
+```
+
+Notice that the `ExtensionData` property name doesn't appear in the JSON. This behavior lets you round-trip the JSON without losing any extra data that otherwise wouldn't be deserialized.
+
+## Use Utf8JsonWriter directly
+
+The following example shows how to use the <xref:System.Text.Json.Utf8JsonWriter> class directly.
+
+```csharp
+var options = new JsonWriterOptions
+{
+    Indented = true
+};
+
+using (var stream = new MemoryStream())
+{
+    using (var writer = new Utf8JsonWriter(stream, options))
+    {
+        writer.WriteStartObject();
+        writer.WriteString("date", DateTimeOffset.UtcNow);
+        writer.WriteNumber("temp", 42);
+        writer.WriteEndObject();
+    }
+
+    string json = Encoding.UTF8.GetString(stream.ToArray());
+    Console.WriteLine(json);
+}
+```
+
+## Use Utf8JsonReader directly
+
+The following example shows how to use the <xref:System.Text.Json.Utf8JsonReader> class directly. The code assumes that the `jsonUtf8` variable is a byte array that contains valid JSON.
+
+```csharp
+Utf8JsonReader reader = new Utf8JsonReader(jsonUtf8, isFinalBlock: true, state: default);
+
+while (reader.Read())
+{
+    Console.Write(reader.TokenType);
+
+    switch (reader.TokenType)
+    {
+        case JsonTokenType.PropertyName:
+        case JsonTokenType.String:
+            {
+                string text = reader.GetString();
+                Console.Write(" ");
+                Console.Write(text);
+                break;
+            }
+
+        case JsonTokenType.Number:
+            {
+                int value = reader.GetInt32();
+                Console.Write(" ");
+                Console.Write(value);
+                break;
+            }
+
+            // Other token types elided for brevity
+    }
+
+    Console.WriteLine();
+}
+```
+
+## Additional resources
```

Comment:
We should consider adding a section on JsonDocument/JsonElement between the serializer/desierlizer and the reader/writer.